demos/noise_suppression_demo/cpp/main.cpp

```diff
@@ -116,131 +117,152 @@ void read_wav(const std::string& file_name, std::vector<int16_t>& wave, RiffWave
 
-        //fp32 input wave will be expanded to be divisible by patch_size
+        std::vector<float> inp_wave_fp32;
+        std::vector<float> out_wave_fp32;
+        // fp32 input wave will be expanded to be divisible by patch_size
         size_t iter = 1 + (inp_wave_s16.size() / patch_size);
         size_t inp_size = patch_size * iter;
         inp_wave_fp32.resize(inp_size, 0);
         out_wave_fp32.resize(inp_size, 0);
 
-        //convert sint16_t  to float
-        float scale = 1.0f/std::numeric_limits<int16_t>::max();
-        size_t i=0;
-        for(; i < inp_wave_s16.size(); ++i) {
+        // convert sint16_t  to float
+        float scale = 1.0f / std::numeric_limits<int16_t>::max();
+        for(size_t i = 0; i < inp_wave_s16.size(); ++i) {
             inp_wave_fp32[i] = (float)inp_wave_s16[i] * scale;
         }
 
         auto start_time = Time::now();
-        for(size_t i=0; i<iter; ++i) {
-            auto inputBlob = InferenceEngine::make_shared_blob<float>(inp_desc, &inp_wave_fp32[i * patch_size]);
-            infer_request.SetBlob(input_name, inputBlob);
+        for(size_t i = 0; i < iter; ++i) {
+            ov::runtime::Tensor input_tensor(ov::element::f32, inp_shape, &inp_wave_fp32[i * patch_size]);
+            infer_request.set_tensor(input_name, input_tensor);
 
-            for (auto &state_name: state_names) {
+            for (auto& state_name: state_names) {
                 const std::string& inp_state_name = state_name.first;
                 const std::string& out_state_name = state_name.second;
 
                 if (i > 0) {
-                    //set input state by coresponding output state from prev infer
-                    auto blob_ptr = infer_request.GetBlob(out_state_name);
-                    infer_request.SetBlob(inp_state_name,blob_ptr);
+                    // set input state by coresponding output state from prev infer
+                    ov::runtime::Tensor state_tensor = infer_request.get_tensor(out_state_name);
+                    infer_request.set_tensor(inp_state_name, state_tensor);
                 } else {
                     // first iteration. set input state to zero tensor.
-                    const InferenceEngine::TensorDesc &state_desc = inputs[inp_state_name]->getInputData()->getTensorDesc();
-                    auto blob_ptr = InferenceEngine::make_shared_blob<float>(state_desc);
-                    blob_ptr->allocate();
-                    for (auto &val: *blob_ptr) {
-                        val = 0;
-                    }
-                    infer_request.SetBlob(inp_state_name,blob_ptr);
+                    ov::Shape state_shape = model->input(inp_state_name).get_shape();
+                    ov::runtime::Tensor state_tensor(ov::element::f32, state_shape);
+                    memset(state_tensor.data<float>(), 0, state_tensor.get_byte_size());
+                    infer_request.set_tensor(inp_state_name, state_tensor);
```

Comment:
Setting tensor should be out of the condition

Reply:
yes, moved out of condition

Reply:
The `infer_request.set_tensor(inp_state_name, state_tensor)` in else block is not removed

Reply:
yeah, good catch. fixed